public/scripts/editor/js/fc/bramble-input-field.js

```diff
@@ -0,0 +1,139 @@
+// This creates an input field that can morph into a static placeholder
+// when a user wants to save their input.
+define(function(require) {
+  var $ = require("jquery");
+
+  function triggerListeners(context, listenerList, args) {
+    if(!listenerList) {
+      return;
+    }
+
+    listenerList.forEach(function(listener) {
+      listener.apply(context, args);
+    });
+  }
+
+  function create(context) {
+    var element;
+
+    if(context.mode === InputField.Modes.STATIC) {
+      element = $("<span></span>");
+    } else {
+      element = $("<input type=\"text\" />");
+    }
+
+    element.attr("id", context.id);
+    element.addClass(context._element.attr("class"));
+
+    return element;
+  }
+
+  function generateElement(mode) {
+    return mode === InputField.Modes.STATIC ?
+           $("<span></span>") : $("<input type=\"text\" />");
+  }
+
+  // This creates a morphable input field and takes in the
+  // container (should be a div as a DOM element or JQuery object)
+  // that will contain the input field. The constructor also takes
+  // a second optional boolean argument that will force the input field
+  // to be added before any other elements in the container (by default,
+  // it is appended to the container)
+  // The input field starts of as a static placeholder
+  function InputField(container, prepend) {
+    // Internal values
+    this._container = container instanceof $ ? container : $(container);
+    this._element = generateElement(InputField.Modes.STATIC);
+    this._listeners = {};
+
+    // Public value
+    this.mode = InputField.Modes.STATIC;
+
+    this._container[prepend ? "prepend" : "append"](this._element);
+  }
+  InputField.Modes = {
+    EDIT: "EDIT",
+    STATIC: "STATIC"
+  };
+  InputField.EventTypes = {
+    SAVE: "SAVE",
+    EDIT: "EDIT"
+  };
+
+  // Morph the field into a static placeholder
+  InputField.prototype.save = function() {
+    var context = this;
+    var value = context.val();
+    var element;
+
+    context.mode = InputField.Modes.STATIC;
+
+    element = create(context);
+    context._element.replaceWith(function() {
+      return element;
+    });
+    context._element = element;
```

Comment:
Apparently so, it seems like the `replaceWith` method creates a new jquery object, so I have to update the reference.